tests/controller/test_app_tree.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+from django.utils.safestring import mark_safe
+
+from controller.app_tree import AppType
+
+
+@pytest.mark.parametrize(
```

Comment:
This is a really big parametrize decorator. Any specific reason? It's kind of hard to read

Reply:
That it is a parametric test: I'm testing the same thing with different inputs. If readability is an issue, I can create several separate tests, but I think this is the point of parametric tests.

Reply:
What I mean is that it's so nested it's almost impossible to tell what is the `app` and what is the `expected`. I couldn't tell you how many different checks this is making.

But it's just a readability thing. The parametric approach makes sense here.